pkgs/tools/networking/envoy/default.nix

```diff
@@ -0,0 +1,116 @@
+{ stdenv, lib, fetchFromGitHub, pkgconfig, buildBazelPackage, git
+, go, autoconf, automake, cmake, libtool, python, ninja
+}:
+
+buildBazelPackage rec {
+  name = "envoy-${version}";
+  version = "1.11.1";
+  rev = "e349fb6139e4b7a59a9a359be0ea45dd61e589c5";
+
+  src = fetchFromGitHub {
+    owner = "envoyproxy";
+    repo = "envoy";
+    rev = "v${version}";
+    sha256 = "09mgb5xlxws8w379wrw1fw5w7adgpsnj5zwa9f1cbwqjdxgzjhg4";
+  };
+
+  nativeBuildInputs = [ autoconf automake cmake git go libtool ninja python ];
+
+  bazelTarget = "//source/exe:envoy";
+
+  dontUseCmakeConfigure = true;
+
+  fetchAttrs = {
+    preBuild = ''
+      patchShebangs .
+
+      # tell rules_go to use the Go binary found in the PATH
+      sed -e 's:go_register_toolchains(go_version = GO_VERSION):go_register_toolchains(go_version = "host"):g' -i WORKSPACE
+    '';
+
+    preInstall = ''
+      # Remove the go_sdk (it's just a copy of the go derivation) and all
+      # references to it from the marker files. Bazel does not need to download
+      # this sdk because we have patched the WORKSPACE file to point to the one
+      # currently present in PATH. Without removing the go_sdk from the marker
+      # file, the hash of it will change anytime the Go derivation changes and
+      # that would lead to impurities in the marker files which would result in
+      # a different sha256 for the fetch phase.
+      rm -rf $bazelOut/external/{go_sdk,\@go_sdk.marker}
+      sed -e '/^FILE:@go_sdk.*/d' -i $bazelOut/external/\@*.marker
+
+      # Remove the gazelle tools, they contain go binaries that are built
+      # non-deterministically. As long as the gazelle version matches the tools
+      # should be equivalent.
+      rm -rf $bazelOut/external/{bazel_gazelle_go_repository_tools,\@bazel_gazelle_go_repository_tools.marker}
+      sed -e '/^FILE:@bazel_gazelle_go_repository_tools.*/d' -i $bazelOut/external/\@*.marker
+
+      # Remove the autoconf caches
+      rm -rf $bazelOut/external/com_github_gperftools_gperftools/autom4te.cache
+      sed -e '/^FILE:@com_github_gperftools_gperftools/autom4te.cache.*/d' -i $bazelOut/external/\@*.marker
+    '';
+
+    sha256 = "1sq8qdbg8f94dbyc607xpq2yrhw61j3yxq6kggywp6fbbg89fh55";
```

Comment:
I'm trying to build this locally to verify that it works, and I'm getting a hash mismatch:

```
$ nix-build -f . -A envoy
...
INFO: All external dependencies fetched successfully.
Loading: 379 packages loaded
Loading: 379 packages loaded
Loading: 379 packages loaded
@nix { "action": "setPhase", "phase": "installPhase" }
installing
hash mismatch in fixed-output derivation '/nix/store/zr2djgq4yra1gyjvgdrmz4qjmfkl3x2m-envoy-1.11.1-deps':
  wanted: sha256:1sq8qdbg8f94dbyc607xpq2yrhw61j3yxq6kggywp6fbbg89fh55
  got:    sha256:0xlgw90m8sk47c6mr561hh7gbczsxlc8cqc2yr8xkv6vgdzydmz5
```
This was after rebasing onto upstream master, before I saw a _different_ hash for  `got:`. Is it possible that not enough is being clean up here or somehow the rule isn't running?

Reply:
It is possible there's other files that need to be cleaned in there.